### ISSUE TITLE
Gamepad support 🎮

### ISSUE THREAD

**Issue by xmacex** [entzmingerc/pit-orchisstra#1]
This could be fun with a gamepad too. I started [a feature branch](https://github.com/xmacex/pit-orchisstra/tree/gamepad) for NES-style controllers with D-pad, A, B, start and select. It is half-tested and there are a few more buttons to map functions to. Would a PR be of interest?

**Comment by entzmingerc** [comment]
Yes! I haven't got an NES-style controller, but I do have an xbox controller, does it use the same code for support? 

I think using START to turn on / off slithering for the selected snake would be useful as well.

**Comment by xmacex** [comment]
Great. As I understand, that controller should work too, perhaps I'll need to add a redundant mapping besides the D-pad, we will see.

I'll do as you suggest, and add speed control on ⬆️ and ⬇️. This matches the existing controls quite well I would think. Let me know if something else would be fun (maybe to add, rather than duplicate the norns key+encoder interface). Well, playtest will inform.

I'll implement those, try a refactor idea which was given to me in a dream, and submit PR for you to see.

**Comment by entzmingerc** [comment]
Sweet! Speed on up/down sounds perfect, I think I did that on the keyboard controls. I'll give it a test soon.

Also if you're digging around in there, line 636 I replaced `player:note_on(snakeNote, 1)` with `player:play_note(snakeNote, 1, 1)` for the nb voices. `play_note` will buffer a call to `note_off` so nb voices like doubledecker won't have hanging notes. I haven't gotten around to pushing this change but I've been using it locally for last couple weeks lol
